Common/Data/Consolidators/TickConsolidator.cs

```diff
@@ -61,14 +63,24 @@ public TickConsolidator(Func<DateTime, CalendarInfo> func)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TickConsolidator"/> class.
+        /// </summary>
+        /// <param name="period">The period.</param>
+        /// <param name="condition">The condition.</param>
+        public TickConsolidator(TimeSpan period, Func<Tick, bool> condition) : base(period)
+        {
+            _condition = condition ?? (t => true);
+        }
+
         /// <summary>
         /// Determines whether or not the specified data should be processd
         /// </summary>
         /// <param name="data">The data to check</param>
         /// <returns>True if the consolidator should process this data, false otherwise</returns>
         protected override bool ShouldProcess(Tick data)
         {
-            return data.TickType == TickType.Trade;
+            return data.TickType == TickType.Trade && _condition(data);
```

Comment:
I think this new `_condition` seems like maybe we need a new derivate of `TickConsolidator` which overrides `ShouldProcess` and calls `base.ShouldProcess(data)`? Same for `TickQuoteBarConsolidator`